XMonad/Util/NamedScratchpad.hs

```diff
@@ -251,9 +282,17 @@ someNamedScratchpadAction f runApp _ns scratchpadName = do
 
             case NE.nonEmpty matchingOnCurrent of
                 -- no matching window on the current workspace -> scratchpad not running or in background
-                Nothing -> case NE.nonEmpty matchingOnAll of
-                    Nothing   -> runApp conf
-                    Just wins -> f (windows . W.shiftWin (W.currentTag winSet)) wins
+                Nothing -> do
+                    -- before we do anything, check for exclusive scratchpads to hide
+                    whenJust (nspExclusives !? name conf) $ \unwanted -> do
+                        let isUnwanted = foldl' (\qs nsp -> qs <||> query nsp) (pure False) unwanted
```

Comment:
Since we really only want the queries of the exclusive scratchpads, it may or may not be preferrable to just store a `Map String (Query Bool)` instead.